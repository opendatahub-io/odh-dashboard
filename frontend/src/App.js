import React from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import {
  Page,
} from '@patternfly/react-core';
import Launcher from './Launcher/Launcher.js'
import Header from './Header.js'

  /**
   * This function wraps up the output from Launcher.js into a React Fragment and returns to the index page
   */
function App() {
  return (
    <React.Fragment>
    <Page
          header={Header}
        >
    <Launcher />      
    </Page>
    </React.Fragment>
      )}

export default App;
