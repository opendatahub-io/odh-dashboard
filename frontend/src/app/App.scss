html, body, #root {
  height: 100%;
}

#dashboard-page-main {
  // Establish positioning relative to the top left of the main container.
  position: relative;
}

.odh-dashboard {
  &__page-content {
    border: 1px solid var(--pf-global--BorderColor--100);
    display: flex;
    flex: 1;
    margin: var(--pf-global--spacer--md);
  }
  .pf-c-app-launcher__group-title {
    font-size: 13px;
  }
  .pf-c-app-launcher__menu-item-external-icon {
    opacity: 1;
    color: var(--pf-global--icon--Color--light);
  }
  &__brand {
    height: 36px;
  }
}
// specificity targeting form elements to override --pf-global--FontSize--md
.pf-c-page, .pf-c-modal-box {
  .pf-c-app-launcher,
  .pf-c-button,
  .pf-c-dropdown,
  .pf-c-dropdown__menu-item,
  .pf-c-dropdown__toggle,
  .pf-c-select,
  .pf-c-select__menu-item,
  .pf-c-select__toggle,
  .pf-c-text-input-group,
  .pf-c-form-control {
    font-size: var(--pf-global--FontSize--sm);
    height: auto;
  }
}
