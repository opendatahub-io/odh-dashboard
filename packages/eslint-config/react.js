module.exports = {
  extends: ['plugin:jsx-a11y/recommended', 'plugin:react/recommended'],

  parserOptions: {
    ecmaFeatures: {
      tsx: true,
      jsx: true,
      useJSXTextNode: true,
    },
  },

  env: {
    browser: true,
  },

  plugins: ['react-hooks'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'jsx-a11y/no-autofocus': ['error', { ignoreNonDOM: true }],
    'jsx-a11y/anchor-is-valid': [
      'error',
      {
        components: ['Link'],
        specialLink: ['to'],
        aspects: ['noHref', 'invalidHref', 'preferButton'],
      },
    ],
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-no-constructed-context-values': 'error',
    'react/no-unused-prop-types': 'error',
    'react/jsx-curly-brace-presence': [2, { props: 'never', children: 'never' }],
    'react/self-closing-comp': 'error',
    'react/prop-types': 'off',
  },
};
