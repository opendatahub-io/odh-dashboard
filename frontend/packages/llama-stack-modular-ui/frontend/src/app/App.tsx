import * as React from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import { BrowserRouter as Router } from 'react-router-dom';
import { AppLayout } from './AppLayout/AppLayout';
import { AppRoutes } from './routes';
import './app.css';

const App: React.FunctionComponent = () => (
  <Router>
    <AppLayout>
      <AppRoutes />
    </AppLayout>
  </Router>
);

export default App;
