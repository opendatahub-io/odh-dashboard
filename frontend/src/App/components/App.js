import React, { useState, useEffect } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";

import { Page } from "@patternfly/react-core";
// import { Header, NavSidebar, Routes } from "./";
import { Header, Routes } from "./";
import { getComponents } from "../actions";

import "./App.scss";

const _App = withRouter(({ getComponents }) => {
  const [isNavOpen, setIsNavOpen] = useState(true);

  useEffect(() => {
    getComponents();
  }, []);

  const onNavToggle = () => {
    setIsNavOpen(!isNavOpen);
  };

  return (
    <Page
      header={<Header isNavOpen={isNavOpen} onNavToggle={onNavToggle} />}
      // sidebar={<NavSidebar isNavOpen={isNavOpen} />}
      isManagedSidebar
      className="app"
    >
      <Routes />
    </Page>
  );
});

const mapStateToProps = (state) => {
  return state.appReducer;
};

const mapDispatchToProps = (dispatch) => ({
  getComponents: () => {
    dispatch(getComponents());
  },
});

export const App = connect(mapStateToProps, mapDispatchToProps)(_App);
export default App;
