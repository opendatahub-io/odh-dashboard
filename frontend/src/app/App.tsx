import React from 'react';
import '@patternfly/patternfly/patternfly.min.css';
import '@patternfly/patternfly/patternfly-addons.css';
import {
  Alert,
  Bullseye,
  Button,
  Page,
  PageSection,
  Spinner,
  Stack,
  StackItem,
} from '@patternfly/react-core';
import ErrorBoundary from '~/components/error/ErrorBoundary';
import ToastNotifications from '~/components/ToastNotifications';
import { useWatchBuildStatus } from '~/utilities/useWatchBuildStatus';
import { useUser } from '~/redux/selectors';
import { DASHBOARD_MAIN_CONTAINER_SELECTOR } from '~/utilities/const';
import useDetectUser from '~/utilities/useDetectUser';
import ProjectsContextProvider from '~/concepts/projects/ProjectsContext';
import Header from './Header';
import AppRoutes from './AppRoutes';
import NavSidebar from './NavSidebar';
import AppNotificationDrawer from './AppNotificationDrawer';
import { AppContext } from './AppContext';
import { useApplicationSettings } from './useApplicationSettings';
import TelemetrySetup from './TelemetrySetup';
import { logout } from './appUtils';

import './App.scss';

const App: React.FC = () => {
  const [notificationsOpen, setNotificationsOpen] = React.useState(false);
  const { username, userError, isAllowed } = useUser();

  const buildStatuses = useWatchBuildStatus();
  const {
    dashboardConfig,
    loaded: configLoaded,
    loadError: fetchConfigError,
  } = useApplicationSettings();

  useDetectUser();

  if (!username || !configLoaded || !dashboardConfig) {
    // We lack the critical data to startup the app
    if (userError || fetchConfigError) {
      // There was an error fetching critical data
      return (
        <Page>
          <PageSection>
            <Stack hasGutter>
              <StackItem>
                <Alert variant="danger" isInline title="General loading error">
                  <p>
                    {(userError ? userError.message : fetchConfigError?.message) ||
                      'Unknown error occurred during startup.'}
                  </p>
                  <p>Logging out and logging back in may solve the issue.</p>
                </Alert>
              </StackItem>
              <StackItem>
                <Button
                  variant="secondary"
                  onClick={() => logout().then(() => window.location.reload())}
                >
                  Logout
                </Button>
              </StackItem>
            </Stack>
          </PageSection>
        </Page>
      );
    }

    // Assume we are still waiting on the API to finish
    return (
      <Bullseye>
        <Spinner />
      </Bullseye>
    );
  }

  return (
    <AppContext.Provider
      value={{
        buildStatuses,
        dashboardConfig,
      }}
    >
      <Page
        className="odh-dashboard"
        isManagedSidebar
        header={<Header onNotificationsClick={() => setNotificationsOpen(!notificationsOpen)} />}
        sidebar={isAllowed ? <NavSidebar /> : undefined}
        notificationDrawer={<AppNotificationDrawer onClose={() => setNotificationsOpen(false)} />}
        isNotificationDrawerExpanded={notificationsOpen}
        mainContainerId={DASHBOARD_MAIN_CONTAINER_SELECTOR}
      >
        <ErrorBoundary>
          <ProjectsContextProvider>
            <AppRoutes />
          </ProjectsContextProvider>
          <ToastNotifications />
          <TelemetrySetup />
        </ErrorBoundary>
      </Page>
    </AppContext.Provider>
  );
};

export default App;
