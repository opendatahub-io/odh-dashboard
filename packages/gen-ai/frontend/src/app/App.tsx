import * as React from 'react';
import { AppLayout } from '~/app/standalone/AppLayout';
import { AppRoutes } from '~/app/AppRoutes';
import '@patternfly/react-core/dist/styles/base.css';
import './app.css';

const App: React.FunctionComponent = () => (
  <AppLayout>
    <AppRoutes />
  </AppLayout>
);

export default App;
