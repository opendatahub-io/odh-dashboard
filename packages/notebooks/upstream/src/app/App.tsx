import * as React from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import AppRoutes from './AppRoutes';
import './app.css';
import {
  Flex,
  Masthead,
  MastheadContent,
  MastheadToggle,
  Page,
  PageToggleButton,
  Title
} from '@patternfly/react-core';
import NavSidebar from './NavSidebar';
import { BarsIcon } from '@patternfly/react-icons';

const App: React.FC = () => {
  const masthead = (
    <Masthead>
      <MastheadToggle>
        <PageToggleButton id="page-nav-toggle" variant="plain" aria-label="Dashboard navigation">
          <BarsIcon />
        </PageToggleButton>
      </MastheadToggle>

      <MastheadContent>
        <Flex>
          <Title headingLevel="h2" size="3xl">
            Kubeflow Notebooks 2.0
          </Title>
        </Flex>
      </MastheadContent>
    </Masthead>
  );

  return (
    <Page
      mainContainerId='primary-app-container'
      masthead={masthead}
      isManagedSidebar
      sidebar={<NavSidebar />}
    >
      <AppRoutes />
    </Page>
  );
};

export default App;