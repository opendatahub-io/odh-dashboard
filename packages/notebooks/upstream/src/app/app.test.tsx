import * as React from "react";
import App from "@app/index";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("App tests", () => {
  test("should render default App component", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render a nav-toggle button", () => {
    render(<App />);

    expect(
      screen.getByRole("button", { name: "Global navigation" }),
    ).toBeVisible();
  });
});
