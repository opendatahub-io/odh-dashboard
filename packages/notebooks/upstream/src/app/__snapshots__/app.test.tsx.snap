// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-page"
  >
    <div
      class="pf-v6-c-skip-to-content"
    >
      <a
        aria-disabled="false"
        class="pf-v6-c-button pf-m-primary"
        data-ouia-component-id="OUIA-Generated-Button-primary-1"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        href="#primary-app-container"
      >
        Skip to Content
      </a>
    </div>
    <header
      class="pf-v6-c-masthead pf-m-display-inline-on-md"
    >
      <span
        class="pf-v6-c-masthead__toggle"
      >
        <button
          aria-disabled="false"
          aria-label="Global navigation"
          class="pf-v6-c-button pf-m-plain"
          data-ouia-component-id="OUIA-Generated-Button-plain-1"
          data-ouia-component-type="PF6/Button"
          data-ouia-safe="true"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </button>
      </span>
      <div
        class="pf-v6-c-masthead__content"
      >
        <div
          class="pf-v6-l-flex"
        >
          <h2
            class="pf-v6-c-title pf-m-3xl"
            data-ouia-component-id="OUIA-Generated-Title-1"
            data-ouia-component-type="PF6/Title"
            data-ouia-safe="true"
          >
            Kubeflow Workspaces
          </h2>
        </div>
      </div>
    </header>
    <div
      aria-hidden="false"
      class="pf-v6-c-page__sidebar pf-m-expanded"
      id="page-sidebar"
    >
      <div
        class="pf-v6-c-page__sidebar-body"
      >
        <nav
          aria-label="Global"
          class="pf-v6-c-nav"
          data-ouia-component-id="OUIA-Generated-Nav-1"
          data-ouia-component-type="PF6/Nav"
          data-ouia-safe="true"
          id="nav-primary-simple"
        >
          <ul
            class="pf-v6-c-nav__list"
            id="nav-list-simple"
            role="list"
          >
            <li
              class="pf-v6-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-1"
              data-ouia-component-type="PF6/NavItem"
              data-ouia-safe="true"
            >
              <a
                aria-current="page"
                class="pf-v6-c-nav__link pf-m-current active"
                href="/"
              >
                Dashboard
              </a>
            </li>
            <li
              class="pf-v6-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-2"
              data-ouia-component-type="PF6/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v6-c-nav__link"
                href="/support"
              >
                Support
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      class="pf-v6-c-page__main-container"
    >
      <main
        class="pf-v6-c-page__main"
        id="primary-app-container"
        tabindex="-1"
      >
        <section
          class="pf-v6-c-page__main-section"
        >
          <div
            class="pf-v6-c-empty-state"
          >
            <div
              class="pf-v6-c-empty-state__content"
            >
              <div
                class="pf-v6-c-empty-state__header"
              >
                <div
                  class="pf-v6-c-empty-state__icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M488.6 250.2L392 214V105.5c0-15-9.3-28.4-23.4-33.7l-100-37.5c-8.1-3.1-17.1-3.1-25.3 0l-100 37.5c-14.1 5.3-23.4 18.7-23.4 33.7V214l-96.6 36.2C9.3 255.5 0 268.9 0 283.9V394c0 13.6 7.7 26.1 19.9 32.2l100 50c10.1 5.1 22.1 5.1 32.2 0l103.9-52 103.9 52c10.1 5.1 22.1 5.1 32.2 0l100-50c12.2-6.1 19.9-18.6 19.9-32.2V283.9c0-15-9.3-28.4-23.4-33.7zM358 214.8l-85 31.9v-68.2l85-37v73.3zM154 104.1l102-38.2 102 38.2v.6l-102 41.4-102-41.4v-.6zm84 291.1l-85 42.5v-79.1l85-38.8v75.4zm0-112l-102 41.4-102-41.4v-.6l102-38.2 102 38.2v.6zm240 112l-85 42.5v-79.1l85-38.8v75.4zm0-112l-102 41.4-102-41.4v-.6l102-38.2 102 38.2v.6z"
                    />
                  </svg>
                </div>
                <div
                  class="pf-v6-c-empty-state__title"
                >
                  <h1
                    class="pf-v6-c-empty-state__title-text"
                  >
                    Empty State (Dashboard Module)
                  </h1>
                </div>
              </div>
              <div
                class="pf-v6-c-empty-state__body"
              >
                <div
                  class="pf-v6-c-content"
                >
                  <p
                    class="pf-v6-c-content--p"
                    data-ouia-component-id="OUIA-Generated-Text-1"
                    data-ouia-component-type="PF6/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    This represents an the empty state pattern in Patternfly 6. Hopefully it's simple enough to use but flexible enough to meet a variety of needs.
                  </p>
                </div>
              </div>
              <div
                class="pf-v6-c-empty-state__footer"
              >
                <button
                  aria-disabled="false"
                  class="pf-v6-c-button pf-m-primary"
                  data-ouia-component-id="OUIA-Generated-Button-primary-2"
                  data-ouia-component-type="PF6/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Primary Action
                </button>
              </div>
            </div>
          </div>
        </section>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
